pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -417,7 +420,14 @@ def select_training_type_plugin(self) -> TrainingTypePlugin:
                 ddp_plugin_cls = DDPShardedPlugin
             elif use_ddp_sharded_spawn:
                 ddp_plugin_cls = DDPSpawnShardedPlugin
-            elif use_ddp_cpu_slurm or use_slurm_ddp or use_ddp_cpu_torch_elastic or use_torchelastic_ddp:
+            elif (
+                    use_ddp_cpu_slurm or
+                    use_slurm_ddp or
+                    use_ddp_cpu_torch_elastic or
+                    use_torchelastic_ddp or
+                    use_kubeflow_ddp or
+                    use_ddp_cpu_kubeflow
+            ):
```

Comment:
This seems like it might be due for a refactor, but I just extended the existing code for now, since I'm not 100% sure of all the edge cases this might be handling.

Reply:
it's getting a bit ridiculous yes. 
I have ideas how to simplify this in the future. But your addition looks solid!
So to summarize: The Kubeflow cluster environment is available with: DDPPlugin, DDPSpawnPlugin, user custom plugin, correct?